File: phone-exchange-2-dot-net-examples/GetInternationalExchangeInfoRestSdkExample.cs
using phone_exchange_2_dot_net.REST;

namespace phone_exchange_2_dot_net_examples
{
    public class GetInternationalExchangeInfoRestSdkExample
    {
        public static void Go(string licenseKey, bool isLive)
        {
            Console.WriteLine("\r\n--------------------------------------------------------");
            Console.WriteLine("PhoneExchange2 - GetInternationalExchangeInfo - REST SDK");
            Console.WriteLine("--------------------------------------------------------");

            GetInternationalExchangeInfoClient.GetInternationalExchangeInfoInput getInternationalExchangeInfoInput = new(
                PhoneNumber: "+12025550123",
                Country: "US",
                LicenseKey: licenseKey,
                IsLive: isLive,
                TimeoutSeconds: 15
            );

            Console.WriteLine("\r\n* Input *\r\n");
            Console.WriteLine($"Phone Number: {getInternationalExchangeInfoInput.PhoneNumber}");
            Console.WriteLine($"Country     : {getInternationalExchangeInfoInput.Country}");
            Console.WriteLine($"License Key : {getInternationalExchangeInfoInput.LicenseKey}");
            Console.WriteLine($"Is Live     : {getInternationalExchangeInfoInput.IsLive}");

            PE2Response response = GetInternationalExchangeInfoClient.Invoke(getInternationalExchangeInfoInput);
            if (response.Error is null)
            {
                Console.WriteLine("\r\n* International Exchange Info *\r\n");
                if (response.InternationalExchangeInfo != null)
                {
                    Console.WriteLine($"Phone Number In     : {response.InternationalExchangeInfo.PhoneNumberIn}");
                    Console.WriteLine($"Country Code        : {response.InternationalExchangeInfo.CountryCode}");
                    Console.WriteLine($"Format National     : {response.InternationalExchangeInfo.FormatNational}");
                    Console.WriteLine($"Extension           : {response.InternationalExchangeInfo.Extension}");
                    Console.WriteLine($"Locality            : {response.InternationalExchangeInfo.Locality}");
                    Console.WriteLine($"Locality Match Level: {response.InternationalExchangeInfo.LocalityMatchLevel}");
                    Console.WriteLine($"Time Zone           : {response.InternationalExchangeInfo.TimeZone}");
                    Console.WriteLine($"Latitude            : {response.InternationalExchangeInfo.Latitude}");
                    Console.WriteLine($"Longitude           : {response.InternationalExchangeInfo.Longitude}");
                    Console.WriteLine($"Country             : {response.InternationalExchangeInfo.Country}");
                    Console.WriteLine($"Country ISO2        : {response.InternationalExchangeInfo.CountryISO2}");
                    Console.WriteLine($"Country ISO3        : {response.InternationalExchangeInfo.CountryISO3}");
                    Console.WriteLine($"Format International: {response.InternationalExchangeInfo.FormatInternational}");
                    Console.WriteLine($"Format E164         : {response.InternationalExchangeInfo.FormatE164}");
                    Console.WriteLine($"Carrier             : {response.InternationalExchangeInfo.Carrier}");
                    Console.WriteLine($"Line Type           : {response.InternationalExchangeInfo.LineType}");
                    Console.WriteLine($"SMS Address         : {response.InternationalExchangeInfo.SMSAddress}");
                    Console.WriteLine($"MMS Address         : {response.InternationalExchangeInfo.MMSAddress}");
                    Console.WriteLine($"Is Valid            : {response.InternationalExchangeInfo.IsValid}");
                    Console.WriteLine($"Is Valid For Region : {response.InternationalExchangeInfo.IsValidForRegion}");
                    Console.WriteLine($"Note Codes          : {response.InternationalExchangeInfo.NoteCodes}");
                    Console.WriteLine($"Note Descriptions   : {response.InternationalExchangeInfo.NoteDescriptions}");
                }
                else
                {
                    Console.WriteLine("No international exchange info items found.");
                }
            }
            else
            {
                Console.WriteLine("\r\n* Error *\r\n");
                Console.WriteLine($"Error Type     : {response.Error.Type}");
                Console.WriteLine($"Error Type Code: {response.Error.TypeCode}");
                Console.WriteLine($"Error Desc     : {response.Error.Desc}");
                Console.WriteLine($"Error Desc Code: {response.Error.DescCode}");
            }
        }
    }
}

File: phone-exchange-2-nodejs-examples/get_international_exchange_info_rest_sdk_example.js
import { GetInternationalExchangeInfoClient } from '../phone-exchange-2-nodejs/REST/get_international_exchange_info_rest.js';

async function GetInternationalExchangeInfoRestGo(licenseKey, isLive) {
    console.log('\n--------------------------------------------------------');
    console.log('PhoneExchange2 - GetInternationalExchangeInfo - REST SDK');
    console.log('--------------------------------------------------------');

    const phoneNumber = '+18059631700';
    const country = 'US';
    const timeoutSeconds = 15;

    console.log('\n* Input *\n');
    console.log(`Phone Number   : ${phoneNumber}`);
    console.log(`Country        : ${country}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const response = await GetInternationalExchangeInfoClient.invoke(phoneNumber, country, licenseKey, isLive, timeoutSeconds);

        if (response.Error) {
            console.log('\n* Error *\n');
            console.log(`Error Type    : ${response.Error.Type}`);
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error TypeCode: ${response.Error.TypeCode}`);
            console.log(`Error DescCode: ${response.Error.DescCode}`);
        }

        console.log('\n* International Exchange Info *\n');
        if (response.InternationalExchangeInfo) {
            console.log(`Phone Number In     : ${response.InternationalExchangeInfo.PhoneNumberIn}`);
            console.log(`Country Code        : ${response.InternationalExchangeInfo.CountryCode}`);
            console.log(`Format National     : ${response.InternationalExchangeInfo.FormatNational}`);
            console.log(`Extension           : ${response.InternationalExchangeInfo.Extension}`);
            console.log(`Locality            : ${response.InternationalExchangeInfo.Locality}`);
            console.log(`Locality Match Level: ${response.InternationalExchangeInfo.LocalityMatchLevel}`);
            console.log(`Time Zone           : ${response.InternationalExchangeInfo.TimeZone}`);
            console.log(`Latitude            : ${response.InternationalExchangeInfo.Latitude}`);
            console.log(`Longitude           : ${response.InternationalExchangeInfo.Longitude}`);
            console.log(`Country             : ${response.InternationalExchangeInfo.Country}`);
            console.log(`Country ISO2        : ${response.InternationalExchangeInfo.CountryISO2}`);
            console.log(`Country ISO3        : ${response.InternationalExchangeInfo.CountryISO3}`);
            console.log(`Format International: ${response.InternationalExchangeInfo.FormatInternational}`);
            console.log(`Format E164         : ${response.InternationalExchangeInfo.FormatE164}`);
            console.log(`Carrier             : ${response.InternationalExchangeInfo.Carrier}`);
            console.log(`Line Type           : ${response.InternationalExchangeInfo.LineType}`);
            console.log(`SMS Address         : ${response.InternationalExchangeInfo.SMSAddress}`);
            console.log(`MMS Address         : ${response.InternationalExchangeInfo.MMSAddress}`);
            console.log(`Is Valid            : ${response.InternationalExchangeInfo.IsValid}`);
            console.log(`Is Valid For Region : ${response.InternationalExchangeInfo.IsValidForRegion}`);
            console.log(`Note Codes          : ${response.InternationalExchangeInfo.NoteCodes}`);
            console.log(`Note Descriptions   : ${response.InternationalExchangeInfo.NoteDescriptions}`);
        } else {
            console.log('No international exchange info results found.');
        }
    } catch (error) {
        console.log('\n* Error *\n');
        console.log(`Error Message: ${error.message}`);
    }
}

export { GetInternationalExchangeInfoRestGo };